### PR TITLE
[cudnn] Support v8 API in fbcode

### PR DESCRIPTION
**Pull request by xw285cornell**
Summary: It turns out we never turn on cudnn v8 API which blocks bf16 conv. Enable the new v8 API

Test Plan: buck run mode/dev-nosan scripts/xdwang/example:fc_pytorch

Reviewed By: ngimel

Differential Revision: D43784279

### CODE REVIEW

**Comment by xw285cornell** [comment]
@ngimel @malfet if you can help take a look :)

File: aten/src/ATen/native/cudnn/Conv_v8.cpp [hunk 1]
```diff
@@ -373,7 +373,7 @@ void generate_and_filter_plans(const cudnnHandle_t handle, cudnn_frontend::Opera
   if (remove_invalid) {
     cudnn_frontend::executionPlans_t new_valid_plans;
     for (auto &plan : valid_plans) {
-      if (plan.getWorkspaceSize() <= max_workspace_size) {
+      if ((size_t) plan.getWorkspaceSize() <= max_workspace_size) {
```
Comment (malfet) [inline aten/src/ATen/native/cudnn/Conv_v8.cpp:376]: Why this change is needed? If it is, then please use `static_cast<size_t>(plan.getWorkspaceSize())`
```suggestion
      if (plan.getWorkspaceSize() <= max_workspace_size) {
```

**Review by malfet — APPROVED** [review]
LGTM, though I'm surprised why `size_t` cast is needed

File: aten/src/ATen/native/cudnn/Conv_v8.cpp (hunk 1 above)
Comment (ngimel) [reply aten/src/ATen/native/cudnn/Conv_v8.cpp:376]: That's our favorite signed-unsigned comparison probably, can we enable compilation flags to error on it in OSS builds? It's been a nuisance.

Comment (ngimel) [reply aten/src/ATen/native/cudnn/Conv_v8.cpp:376]: Given that `getWorkspaceSize()` returns `int64_t` in cudnn frontend, a better fix would be to change our vars like max_worksplace_size and curr_workspace_size to int64_t instead of size_t

Comment (xw285cornell) [reply aten/src/ATen/native/cudnn/Conv_v8.cpp:376]: sounds good! I'll make the change and see if I can make this from warning to error in oss :)

**Comment by xw285cornell** [comment]
Looks this PR has a conflict with https://github.com/pytorch/pytorch/pull/96723, will probably wait a bit to get in sync
